Best Practices guide update (#334)
astronomer/airflow-guides: guides/dag-best-practices.md

</ul>

## Leverage Airflow Features and Ecosystem

The next category of best practices relates to getting the most out of Airflow by leveraging built-in features and the broader Airflow ecosystem to tailor your DAGs to your specific use case. Using Airflow in this way makes it easier to scale and pull in the right tools based on your needs.

### Make Use of Provider Packages

A great feature of Airflow is its [provider packages](https://airflow.apache.org/docs/apache-airflow-providers/), which orchestrate jobs using third party tools. One of the best aspects of Airflow is its robust and active community, which has resulted in many integrations between Airflow and other tools in the data ecosystem. Wherever possible, it's best practice to make use of these integrations rather than writing Python functions yourself (no need to reinvent the wheel). This makes it easier for teams using existing tools to adopt Airflow, and it means you get to write less code since many existing hooks and operators have taken care of that for you.

For easy discovery of all the great provider packages out there, check out the [Astronomer Registry](https://registry.astronomer.io/).

> ### Be Intentional About Data Processing

Reviewer: I think the key with this header is to be more specific about the recommendation. I gathered that this is the key point about the section, but it could also easily be `Set Up Airflow for Data Processing Jobs`, since you focus on that a bit more towards the end.

> Since DAGs are written in Python, you have many options available for implementing data processing. For small to medium scale workloads, it is typically safe to do your data processing within Airflow as long as you allocate enough resources to your Airflow infrastructure. On the other hand, processing very large amounts of data is typically best offloaded to a framework specifically optimized for those use cases, like [Apache Spark](https://spark.apache.org/). You can then use Airflow to orchestrate those jobs.

Reviewer: ```suggestion
Because DAGs are written in Python, you have many options available for implementing data processing. For small to medium scale workloads, it is typically safe to do your data processing within Airflow as long as you allocate enough resources to your Airflow infrastructure. Large data processing jobs are typically best offloaded to a framework specifically optimized for those use cases, such as [Apache Spark](https://spark.apache.org/). You can then use Airflow to orchestrate those jobs.
```
